test/xml/test_schema.rb

```diff
@@ -171,6 +175,10 @@ class TestNokogiriXMLSchema < Nokogiri::TestCase
 
         assert(errors = xsd.validate(tempfile.path))
         assert_equal(2, errors.length)
+        assert_equal(
+          ["/purchaseOrder/billTo/state", "/purchaseOrder/shipTo/state"],
+          errors.map(&:path).sort,
+        )
```

Comment:
Let's change this back to asserting that the paths are nil, since the file parsing case won't have a DOM node to refer to, and please document that for the `#path` attribute.